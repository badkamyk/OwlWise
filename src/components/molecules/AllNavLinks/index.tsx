import NavLink from "../../atoms/NavLink"

export default function AllNavLinks({ isExpanded }: { isExpanded: boolean }) {
    const navLinks = ["Home", "About", "Services", "Contact"]
    const navLinksList = navLinks.map((link) => {
        return <NavLink key={link + " link"} href={link} />
    })
    const navLinksClass = isExpanded ? "block" : "hidden"

    return (
        <div className={`${navLinksClass} w-full md:block md:w-auto`} id="navbar-default">
            <ul className="flex flex-col p-4 mt-4 border border-gray-100 rounded-lg bg-gray-50 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 md:bg-white dark:bg-gray-800 md:dark:bg-gray-900 dark:border-gray-700">
                {navLinksList}
            </ul>
        </div>
    );
}
