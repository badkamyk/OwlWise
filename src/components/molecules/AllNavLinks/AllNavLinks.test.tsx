import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import AllNavLinks from "./index";

it("renders AllNavLinks correctly", () => {
	const { container } = render(<AllNavLinks isExpanded={true} />);
	expect(screen.getByText("Home")).toBeTruthy();
	expect(screen.getByText("About")).toBeTruthy();
	expect(screen.getByText("Pricing")).toBeTruthy();
	expect(screen.getByText("Contact")).toBeTruthy();
	expect(container).toMatchSnapshot();
});
