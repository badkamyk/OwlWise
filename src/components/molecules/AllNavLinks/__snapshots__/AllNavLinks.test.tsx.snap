// Vitest Snapshot v1

exports[`AllNavLinks 1`] = `
<div>
  <div
    class="block w-full md:block md:w-auto"
    id="navbar-default"
  >
    <ul
      class="mt-4 flex flex-col rounded-lg border border-gray-100 bg-gray-50 p-4 dark:border-gray-700 dark:bg-gray-800 md:mt-0 md:flex-row md:space-x-8 md:border-0 md:bg-white md:text-sm md:font-medium md:dark:bg-gray-900"
    >
      <li>
        <a
          aria-current="page"
          class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
          href="/pricing"
        >
          Pricing
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders AllNavLinks correctly 1`] = `
<div>
  <div
    class="block w-full md:block md:w-auto"
    id="navbar-default"
  >
    <ul
      class="mt-4 flex flex-col rounded-lg border border-gray-100 bg-gray-50 p-4 dark:border-gray-700 dark:bg-gray-800 md:mt-0 md:flex-row md:space-x-8 md:border-0 md:bg-white md:text-sm md:font-medium md:dark:bg-gray-900"
    >
      <li>
        <a
          aria-current="page"
          class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
          href="/pricing"
        >
          Pricing
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="block rounded bg-blue-700 py-2 pl-3 pr-4 text-white dark:text-white md:bg-transparent md:p-0 md:text-blue-400"
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </div>
</div>
`;
